Allow drag-clicking before symbol lookup

Jummit opened the pull request:
This makes it possible for touch and graphics tablet users to do symbol lookups in the code editor by ctrl+clicking on a symbol. (Previously this wasn't possible because drag events invalidated the click).

~~As a consequence, starting a selection, hitting ctrl and releasing the mouse over a symbol triggers the lookup. Please let me know if this behavior is considered a bug, so that I can find a cleaner solution (probably involving some sort of maximum distance from the selection start).~~

<details>

![demonstration](https://user-images.githubusercontent.com/28286961/177043433-331d68e0-c285-423b-a5cd-990159603531.gif)

</details>

Paulb23 reviewed (changes requested):
> Please let me know if this behavior is considered a bug

Yeah this introduces a bug, see #32232

Jummit commented:
> > Please let me know if this behavior is considered a bug
> 
> Yeah this introduces a bug, see #32232

I don't think this is the same bug. I can't replicate selection or clicking on the minimap to open the documentation. Can you elaborate on how to replicate the bug?

Paulb23 commented:
Apologies should have been clearer, I meant releasing the mouse at the end of a selection going to help would be a bug.

Jummit commented:
I added a fix for the unwanted behavior: The lookup is only canceled when the user drag-clicks more than one character.

Jummit commented:
Whoops, that was the wrong branch. Sorry!

Jummit committed:
Allow drag-clicking before symbol lookup

This makes it easier for touch and tablet users to do symbol lookups in
the code editor.

Jummit commented:
I resolved the conflict, this should be ready to merge.

akien-mga commented on scene/gui/code_edit.cpp:
```diff
@@ -371,12 +371,15 @@ void CodeEdit::gui_input(const Ref<InputEvent> &p_gui_input) {
 					emit_signal(SNAME("symbol_validate"), symbol_lookup_new_word);
 				}
-			} else {
+			} else if (!mm->is_command_or_control_pressed()) {
+				set_symbol_lookup_word_as_valid(false);
+			} else if (mm->get_button_mask() != MouseButton::NONE && symbol_lookup_pos != get_line_column_at_pos(mpos)) {
 				set_symbol_lookup_word_as_valid(false);
```
Both branches do the same so they could be chained with `||`.

Jummit replied:
Makes sense.

Jummit commented:
This is still desired, if anyone wants to take a look at it.

akien-mga commented:
Thanks!
